x-pack/legacy/plugins/maps/server/maps_telemetry/maps_telemetry.ts

```diff
@@ -10,22 +10,18 @@ import {
   ES_GEO_FIELD_TYPE,
   MAP_SAVED_OBJECT_TYPE,
   TELEMETRY_TYPE,
+  // @ts-ignore
 } from '../../common/constants';
 
-function getSavedObjectsClient(server) {
-  const { SavedObjectsClient, getSavedObjectsRepository } = server.savedObjects;
-  const callCluster = server.plugins.elasticsearch.getCluster('admin').callWithInternalUser;
-  const internalRepository = getSavedObjectsRepository(callCluster);
-  return new SavedObjectsClient(internalRepository);
-}
-
-function getUniqueLayerCounts(layerCountsList, mapsCount) {
+function getUniqueLayerCounts(layerCountsList: any[], mapsCount: number) {
```

Comment:
Let's try avoiding the use of `any` since we're doing these TypeScript conversions anyway.

We can introduce a type-definition here, or use `unknown`

Reply:
As a result of this update, there was one case below where I couldn't easily use an `interface` or `unknown` and had to fall back to `any`: `tCounts: any`. If this is a sticking point, we can take another look.

Reply:
So I think we can actually try and get rid of these `any`-types, and not resort to `unknown` (there's very little unknown, since we reference either typed typescript-code from Kibana-core or code from Maps).

I made a suggestion here of how this would look for this PR: https://github.com/elastic/kibana/pull/57508/files#diff-6f8221805f81e4dd616157afd0acb744

imho, for this PR, there is basically three categories of types we need to handle:

- types internal to this module:  `IStats`, ... We can fully define those internally in this module.
- types from the maps-app: `ILayerDescriptor`, `ISourceDescriptor`, etc... This PR is a good place to start introducing type-definitions for those. Since in Maps, descriptors are both referenced server and public-side, we can put them in `common`. Let's just start with the minimum we need. This can grow as we continue to write more TypeScript code.
- Internal saved-object types (e.g. `IndexPatternSavedObjects`, ...) : separate plugins seem to define these ad-hoc to avoid using any/unknown (e.g. Lens has separate definition). We can do a minimal shim for those as well.

Reply:
Got it. What needed to happen was just to make all of the types "agree". The type of the array passed in didn't match the iterator type in the reducer. Wrt the rest of the fie types, I've also added in the require interfaces and removed types that just aren't required here. No `any`s left!

Reply:
I implemented the suggested changes with the exception of the `IndexPatternSavedObject` where I used `IIndexPattern` from the data plugin. Good suggestions! Let me know if anything appears to be missing.